### DISCUSSION TITLE
Test Discussion

### DISCUSSION THREAD

**Discussion by 0xhung295** [Cyfrin/foundry-full-course-cu#2794] [General]
just for testing

**Comment by EngrPips** [comment]
Hello Test

**Comment by 0xhung295** [comment]
Hi, I'm getting an Error when trying to compile on zKsync Plugin although I used Solidity version of 0.8.24

![image](https://github.com/user-attachments/assets/e745a935-ad01-4d88-a0b8-51058cfea86c)

'''
Failed to compile:
Error HH606: The project cannot be compiled, see reasons below.
The Solidity version pragma statement in these files doesn't match any of the configured compilers in your config. Change the pragma or configure additional compiler versions in your hardhat config.
'''

Here is my code:
'''solidity
//SPDX-License-Identifier: MIT
pragma solidity 0.8.24;
library PriceConverter{
    function getPrice() internal view returns(uint256) {
        // address: 
        //contracts to getConversionRateet ETH/USD-Sphophie Testnet: 0x694AA1769357215DE4FAC081bf1f309aDC325306
        //contracts to getConversionRateet ETH/USD-zKsyncSphophie Testnet: 0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF
        //ABI
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF);
        (, int256 price,,,)=priceFeed.latestRoundData();
        //Price of ETH in termsof USD
        return uint256(price* 1e10);
    }
    function getConversionRate(uint256 ethAmount) internal view returns(uint256)  {
        uint256 ethPrice=getPrice();
        uint256 ethAmountUsd =(ethPrice*ethAmount)/1e18;
        return ethAmountUsd;
    }
    function getVersion() internal view returns(uint256) {
        return AggregatorV3Interface(0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF).version();
    }
}
'''

**Reply by EngrPips** [reply]
Put your file in the contracts folder then retry.

**Comment by 0xhung295** [comment]
'''solidity

        // //Transfer
        // payable(msg.sender).transfer(address(this).balance);
        // //send
        // bool sendSuccess=payable(msg.sender).send(address(this).balance);
        // require(sendSuccess, "Send failed");
        // //call
        (bool callSuccess,)=payable(msg.sender).call{value:address(this).balance}("");
        require(callSuccess,"Call failed");
'''

**Comment by EngrPips** [comment]
Hmmmm 🤔
